src/core/ext/transport/chaotic_good/client_transport.h

```diff
@@ -74,6 +74,25 @@ class ClientTransport {
       reader_.reset();
     }
   }
+  void AbortWithError() {
+    // Mark transport as unavailable when the endpoint write/read failed.
+    // Close all the available pipes.
+    if (!outgoing_frames_.IsClosed()) {
+      outgoing_frames_.MarkClosed();
+    }
+    std::map<uint32_t, std::shared_ptr<InterActivityPipe<
+                           ServerFrame, server_frame_queue_size_>::Sender>>
+        stream_map;
+    {
+      MutexLock lock(&mu_);
+      stream_map = stream_map_;
```

Comment:
This creates a copy of the stream_map_ and then iterates through every entry. Why not just directly iterate over the original stream_map_ and mark each stream as Closed under the lock ?

Reply:
Yeah, thanks for catching this. I moved the iterate under the lock now.